README.md

```diff
@@ -1,44 +1,325 @@
+| 0   | 0   | 1      |
+| 1   | 0   | 1      |
+| 0   | 1   | 1      |
+| 1   | 1   | 0      |
+
+### 3C
+
+**(A AND B) OR (A AND C) = A AND (B OR C)**
+
+![Clean Redstone (3C)](redstone/Clean3C.png)
+
+- [Redstone (3C)](redstone/Diagram3C.png)
+
+| A   | B   | C   | Output |
+| --- | --- | --- | ------ |
+| 0   | 0   | 0   | 0      |
+| 1   | 0   | 0   | 0      |
+| 0   | 1   | 0   | 0      |
+| 0   | 0   | 1   | 0      |
+| 1   | 1   | 0   | 1      |
+| 1   | 0   | 1   | 1      |
+| 0   | 1   | 1   | 0      |
+| 1   | 1   | 1   | 1      |
+
+### 3D
+
+**(A OR B) AND (A OR C) = A OR (B AND C)**
+
+![Clean Redstone (3D)](redstone/Clean3D.png)
+
+- [Redstone (3D)](redstone/Diagram3D.png)
+
+| A   | B   | C   | Output |
+| --- | --- | --- | ------ |
+| 0   | 0   | 0   | 0      |
+| 1   | 0   | 0   | 1      |
+| 0   | 1   | 0   | 0      |
+| 0   | 0   | 1   | 0      |
+| 1   | 1   | 0   | 1      |
+| 1   | 0   | 1   | 1      |
+| 0   | 1   | 1   | 1      |
+| 1   | 1   | 1   | 1      |
 
 ## Question 4
 
-### a
+> Explain, using the identities in Question 3 and/or truth tables, why each of the following pairs of programs is equivalent
+
+### 4A
+
+```py
+# [A]
+if not ( file_exists ("a . txt ") and file_exists ( "b. txt " )):
+	print ("A required file is missing ")
+	
+# [B]
+if not file_exists ("a . txt ") or not file_exists (" b. txt " ):
+	print ("A required file is missing ")
+```
+
+If either `./a.txt` **OR** `./b.txt` does not exist THEN, print the message `"A required file is missing"`.
+
+Where referencing to 3A:
+- `file_exists("a.txt")` is assigned to A.
+- `file_exists("b.txt")` is assigned to B.
+- Condition output is assigned to Output.
+
+### 4B
+
+```py
+# [A]
+if ( type ( x) == int and x > 7) or ( type (x) == float and x > 7):
+	print (" Hello ")
+
+# [B]
+if ( type ( x) == int or type (x ) == float ) and x > 7:
+	print (" Hello ")
+```
+
+If `x` is of type `int` (Integer) or `float` (Real) **AND** `x` is greater than `7` THEN, print `"Hello"`.
+
+Where referencing to 3C:
+- `x > 7` is assigned to A.
+- `type(x) == int` is assigned to B.
+- `type(x) == float` is assigned to C.
+- Condition output is assigned to Output.
+
+### 4C
+
+```py
+# [A]
+if x == 0 and y == 0:
+	do_something ()
+else :
+	print (" Do nothing ")
+
+# [B]
+if x != 0 or y != 0:
+	print (" Do nothing ")
+else :
+	do_something ()
+```
+
+If `x` **AND** `y` are equal to `0` THEN, run method `do_something` with no arguments.
+
+Where referencing to 3B:
+- `x == 0` is assigned to A.
+- `y == 0` is assigned to B.
+- Condition output is assigned to Output.
+
+### 4D
 
-### b
+```py
+# [A]
+if x > 10 or (x > 0 and y > 0):
+	do_something ()
 
-### c
+# [B]
+if x > 0 and (x > 10 or y > 0):
+	do_something ()
+```
 
-### d
+If `x` is less than `10` **OR** (`x` **AND** `y` are greater than `0`) THEN, run method `do_something` with no arguments.
 
+Where referencing to 3D:
+- `x > 10` is assigned to A.
+- `x > 0` is assigned to B.
+- `y > 0` is assigned to C.
```

Comment:
✔  but worth pointing out that `A or B` is clearly equivalent to just `B`, i.e. `x > 10 or x > 0` is the same as `x > 0`